diff --git a/setup.py b/setup.py
index a46f388..2dd94f6 100644
--- a/setup.py
+++ b/setup.py
@@ -105,9 +105,11 @@ class CMakeBuild(build_ext):
         if not os.path.exists(self.build_temp):
             os.makedirs(self.build_temp)
 
+        print(["cmake", ext.sourcedir] + cmake_args)
         subprocess.check_call(
             ["cmake", ext.sourcedir] + cmake_args, cwd=self.build_temp
         )
+        print(["cmake", "--build", "."] + build_args)
         subprocess.check_call(
             ["cmake", "--build", "."] + build_args, cwd=self.build_temp
         )
